src/main/src/test/java/org/geoserver/ows/StylePublisherTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+package org.geoserver.ows;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.platform.GeoServerResourceLoader;
+import org.geoserver.test.GeoServerSystemTestSupport;
+import org.junit.Test;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.mock.web.MockServletContext;
+
+/**
+ *
+ * @author Ian Schneider <ischneider@boundlessgeo.com>
+ */
+public class StylePublisherTest extends GeoServerSystemTestSupport {
+
+    static StylePublisher publisher;
+
+    static List<String[]> paths = new ArrayList<String[]>();
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+
+        Catalog catalog = getCatalog();
+        publisher = new StylePublisher(catalog);
+        publisher.setServletContext(new MockServletContext());
+        GeoServerResourceLoader resourceLoader = getResourceLoader();
+
+        // add style - global
+        resourceLoader.copyFromClassPath("org/geoserver/ows/smileyface.png",
+                "styles/smileyface.png");
+        paths.add(new String[] { "styles", "smileyface.png" });
+
+        // add style - workspaced
+        resourceLoader.copyFromClassPath("org/geoserver/ows/house.svg",
+                "workspaces/cite/styles/house.svg");
+        paths.add(new String[] { "styles", "cite", "house.svg" });
```

Comment:
@jodygarnett Yes while it is not yet implemented in a test case, it will be useful in future stages when we ensure that only the correct file types (ie. *.png, *.svg) are accepted (rather than *.xml for example).